### PR TITLE
Add \Fimpebreak, to support of implicit ebreak.

### PR DESCRIPTION
**Pull request by timsifive**
Issue #160.

### CODE REVIEW

File: debug_module.tex [hunk 1]
```diff
@@ -242,18 +242,24 @@ \section{Program Buffer} \label{programbuffer}
...
 execute it exactly once with the Access Register Abstract Command,
-
 setting the \Fpostexec bit in \Rcommand.
-If \Fprogbufsize is 1, the Program Buffer may only hold a single instruction.
-This can be a 32-bit
+The debugger can write whatever program it likes (including jumps out of the
+Program Buffer), but the program must end with
+{\tt ebreak} or {\tt ebreak.c}. To save hardware, an implementation may support
+an implied {\tt ebreak} that is executed when a hart runs off the end of the
+Program Buffer. With this feature, a Program Buffer of just 2 32-bit words can
```
Comment (mwachs5) [inline debug_module.tex]: shouldn't we say "as indicated by \Fimpebreak" or something similar?

File: xml/dm_registers.xml [hunk 2]
```diff
@@ -17,7 +17,14 @@
         time.  Eg. in a multi-hart system some might temporarily be powered
         down, or a system might support hot-swapping harts.
 
-        <field name="0" bits="31:21" access="R" reset="0" />
+        <field name="0" bits="31:23" access="R" reset="0" />
```
Comment (mwachs5) [inline xml/dm_registers.xml]: "reset" should not be 0, it should be "whatever it is".

File: xml/dm_registers.xml [hunk 3]
```diff
@@ -17,7 +17,14 @@
...
+        <field name="impebreak" bits="22" access="R" reset="Preset">
+            If 1, then there is an implicit {\tt ebreak} instruction that will
+            be executed if execution runs off the end of the Program Buffer.
```
Comment (mwachs5) [inline xml/dm_registers.xml]: can we make this a bit more clear... If program buffer word [\Fprogbufsize -1] is executed. To make it clear what is "the end" of program buffer.

File: xml/dm_registers.xml (hunk 2 above)
Comment (timsifive) [reply xml/dm_registers.xml]: It's 0 for the 0 part of the register. For the impebreak it's "Preset" (see one line down).

File: debug_module.tex (hunk 1 above)
Comment (timsifive) [reply debug_module.tex]: Good idea.

File: xml/dm_registers.xml (hunk 3 above)
Comment (timsifive) [reply xml/dm_registers.xml]: I started down that route, but gave up when I couldn't think of a good phrasing for:
After the last instruction in the Program Buffer is executed, and does not change the PC in any way besides normal incrementing...

Suggestions?

File: xml/dm_registers.xml (hunk 2 above)
Comment (mwachs5) [reply xml/dm_registers.xml]: ah yep.

File: xml/dm_registers.xml (hunk 3 above)
Comment (mwachs5) [reply xml/dm_registers.xml]: Hmm, how about just saying there is an implicit {\tt ebreak} at the (nonexistent) ProgBuf location ProgBuf[\Fprogbufsize]. That's really what this is saying.

**Comment by timsifive** [comment]
How's this?

**Pull request merged**